projects/online/online/utils/searcher.py

```diff
@@ -0,0 +1,181 @@
+    return float(tconvert(datetime.fromtimestamp(timestamp, tz=timezone.utc)))
+
+
+@dataclass
+class Event:
+    gpstime: float
+    detection_statistic: float
+    far: float
+    ifos: List[str]
+    datadir: Path
+    ifo_suffix: Optional[str] = None
+
+    def get_frame_write_time(self):
+        """
+        Get the time the frame corresponding
+        to this event was written to disk
+        """
+        t_write = 0
+        if self.ifo_suffix is not None:
+            ifo_dir = "_".join([self.ifos[0], self.ifo_suffix])
+        else:
+            ifo_dir = self.ifos[0]
+        prefix, length, _ = get_prefix(self.datadir / ifo_dir)
+        middle = "_".join(prefix.split("_")[1:])
+        for ifo in self.ifos:
+            if self.ifo_suffix is not None:
+                ifo_dir = "_".join([ifo, self.ifo_suffix])
+            else:
+                ifo_dir = ifo
+            prefix = f"{ifo[0]}-{ifo}_{middle}"
+            fname = (
+                self.datadir
+                / ifo_dir
+                / f"{prefix}-{int(self.gpstime)}-{length}.gwf"
+            )
+            t_write = max(t_write, os.path.getmtime(fname))
+
+        return gps_from_timestamp(t_write)
+
+    def __str__(self):
+        return (
+            "Event("
+            f"gpstime={self.gpstime:0.3f}, "
+            f"detection_statistic={self.detection_statistic:0.2f}, "
+            f"far={self.far:0.3e} Hz"
+            ")"
+        )
+
+    @property
+    def filename(self):
+        return f"event_{int(self.gpstime)}.json"
+
+    def write(self, directory: Path):
+        """
+        Write event information to a local directory
+        """
+        path = directory / f"event_{int(self.gpstime)}"
+        path.mkdir(exist_ok=True, parents=True)
+        filename = path / f"event-{int(self.gpstime)}.json"
+
+        event = asdict(self)
+        filecontents = str(event)
+        filecontents = json.loads(filecontents.replace("'", '"'))
+        with open(filename, "w") as f:
+            json.dump(filecontents, f)
+
+        return path
+
+
+class Searcher:
+    """
+    Object for managing aframe search state, building aframe events,
+    """
+
+    def __init__(
+        self,
+        background: EventSet,
+        far_threshold: float,
+        inference_sampling_rate: float,
+        refractory_period: float,
+    ) -> None:
+        self.inference_sampling_rate = inference_sampling_rate
+        self.refractory_period = refractory_period
+
+        # initialize the state of the searcher:
+
+        # flag that declares if we're in the process
+        # of detecting an event between frames
+        self.detecting = False
+
+        self.last_detection_time = time.time() - self.refractory_period
+
+        # calculate the detection statistic threshold
+        # corresponding to the requested FAR threshodl
+        self.background = background
+        self.threshold = self.background.threshold_at_far(far_threshold)
```

Comment:
The FAR threshold needs to be converted to Hz before calling this, it's currently in 1/days

Reply:
Yeah this is another thing I was thinking about: I think we should migrate to only using Hz when defining far thresholds.

Reply:
Agreed